src/components/query-selector/measure-file-upload.tsx

```diff
@@ -0,0 +1,169 @@
+import { measureBundleState } from '@/state/measure-bundle';
+import { Center, Text } from '@mantine/core';
+import { Dropzone } from '@mantine/dropzone';
+import { showNotification } from '@mantine/notifications';
+import { IconAlertCircle, IconFileCheck, IconFileImport } from '@tabler/icons-react';
+import { bulkQueries, group } from 'fqm-bulk-utils';
+import { useState } from 'react';
+import { useRecoilState } from 'recoil';
+
+export default function MeasureFileUpload() {
+  // const [dropStatus, setDropStatus] = useState<'idle' | 'accept' | 'reject'>('idle');
+  const [loading, setLoading] = useState<boolean>();
+  const [measureBundle, setMeasureBundle] = useRecoilState(measureBundleState);
+
+  const updateDropStatus = (status: 'idle' | 'accept' | 'reject') => {
+    setMeasureBundle(mb => ({
+      ...mb,
+      status: status
+    }));
+    setLoading(false);
+  };
+
+  const rejectUpload = (message: string) => {
+    updateDropStatus('reject');
+    showNotification({
+      icon: <IconAlertCircle />,
+      title: 'Bundle upload failed',
+      message: `There was an issue uploading your bundle: ${message}`,
+      color: 'red'
+    });
+  };
+
+  const handleFileRead = (file: File) => {
+    const reader = new FileReader();
+    reader.onload = () => {
+      try {
```

Comment:
When you initially upload a bad file, it doesn't have the filename. We need to do an initial set of the filename in the measure bundle state so that we can populate that in the reject case. Potential fix:
```suggestion
    reader.onload = () => {
      setMeasureBundle(mb => ({
        ...mb,
        fileName: file.name,}));
      try {
```